src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksNewVendorRequestsServiceImpl.java

```diff
@@ -351,17 +354,37 @@ private PaymentWorksVendor setStatusValuesForPaymentWorksProcessedNewVendorRejec
         return pmwVendor;
     }
     
-    private boolean pmwNewVendorSaveToStagingTableWasSuccessful(PaymentWorksVendor pmwNewVendorRequestModifiedByOjbSave, PaymentWorksNewVendorRequestsBatchReportData reportData) {
+    private boolean pmwNewVendorSaveToStagingTableWasSuccessful(PaymentWorksVendor pmwNewVendorRequestModifiedByOjbSave, 
+            PaymentWorksVendor pmwNewVendorRequestReceivedFromWebServiceCall, String processingErrorMessage, PaymentWorksNewVendorRequestsBatchReportData reportData) {
         if (ObjectUtils.isNotNull(pmwNewVendorRequestModifiedByOjbSave)) {
             return true;
-        }
-        else {
+        } else {
             reportData.getRecordsThatCouldNotBeProcessedSummary().incrementRecordCount();
-            reportData.addPmwVendorThatCouldNotBeProcessed(new PaymentWorksBatchReportRawDataItem(pmwNewVendorRequestModifiedByOjbSave.toString(), getConfigurationService().getPropertyValueAsString(PaymentWorksKeyConstants.INITIAL_SAVE_TO_PMW_STAGING_TABLE_FAILED_ERROR_MESSAGE)));
+            reportData.addPmwVendorThatCouldNotBeProcessed(new PaymentWorksBatchReportRawDataItem(pmwNewVendorRequestReceivedFromWebServiceCall.toString(), 
+                    MessageFormat.format(getConfigurationService().getPropertyValueAsString(PaymentWorksKeyConstants.INITIAL_SAVE_TO_PMW_STAGING_TABLE_FAILED_ERROR_MESSAGE), processingErrorMessage)));
             return false;
         }
     }
     
+    private String parseSqlExceptionStringForOracleError(String exceptionString) {
+        int startPosition = exceptionString.indexOf(PaymentWorksConstants.PaymentWorksVendorStagingTableParseExceptionConstants.EXCEPTION_REASON_START_STRING);
+        int endPosition = exceptionString.indexOf(PaymentWorksConstants.PaymentWorksVendorStagingTableParseExceptionConstants.EXCEPTION_REASON_END_CHAR, startPosition);
+        String reasonForSqlException = PaymentWorksConstants.PaymentWorksVendorStagingTableParseExceptionConstants.DEFAULT_EXCEPTION_MESSAGE;
+        
+        if (requestedIndexWasFound(startPosition) && requestedIndexWasFound(endPosition)) {
+            reasonForSqlException = exceptionString.substring(startPosition, (endPosition + 1));
+        }
```

Comment:
do you think it would be helpful to add an else clause logging that the error message couldn't be parsed?

Reply:
Getting the default error message in the report implies that the error message could not be parsed. 
Do we need the additional noise in the logs since the stacktrace will be in the log from the Exception catch block?

Reply:
fair enough, lets go with as is